### PR TITLE
[explorer] fix: Restriction display issue

### PR DESCRIPTION
**Pull request by AnthonyLaw**
## What was the issue?
- some formatting issues cause the UI to display data incorrectly.

image

## What's the fix?
- fix and refactor for `RestrictionField` and `RestrictionService`
- added unit test

`/mosaics/1788BA84888894EB`
image

`/accounts/TA4RYHMNHCFRCT2PCWOCJMWVAQ3ZCJDOTF2SGBI`
image

### CODE REVIEW

File: __tests__/infrastructure/RestrictionService.spec.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+			it(`returns mosaic global restriction dto with type ${restrictionType.name} and without reference mosaic`, () => {
+				// Arrange:
+				const mosaicGlobalRestriction = createMosaicGlobalRestriction(new MosaicId('0'.repeat(16)));
+
+				// Act:
+				const result = RestrictionService.formatMosaicGlobalRestriction(mosaicGlobalRestriction);
+
+				// Assert:
+				expect(result).toEqual({
+					compositeHash: mosaicGlobalRestriction.compositeHash,
```
Comment (Jaguar0625) [inline __tests__/infrastructure/RestrictionService.spec.js:33]: title: without reference => with self reference?

or is test wrong in that object contains referenceMosaicId

File: __tests__/infrastructure/RestrictionService.spec.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+				restrictionType: {
+					type: MosaicRestrictionType.GE,
+					name: 'great equal'
+				},
+				expectedResult: 'mosaicRestrictionType.GE'
+			},
+			{
+				restrictionType: {
+					type: MosaicRestrictionType.GT,
+					name: 'greater than'
```
Comment (Jaguar0625) [inline __tests__/infrastructure/RestrictionService.spec.js:88]: do you intend to mix great and greater ?

**Comment by cryptoBeliever** [comment]
@AnthonyLaw I got a question regarding global restriction. You added mosaicId to the "Restrictions" column:

image

it's possible that we will have another mosaicId that we currently looking at the page?

File: __tests__/infrastructure/RestrictionService.spec.js (hunk 1 above)
Comment (AnthonyLaw) [reply __tests__/infrastructure/RestrictionService.spec.js:33]: > without reference => with self-reference?

yes you are right. this is better name.

File: __tests__/infrastructure/RestrictionService.spec.js (hunk 2 above)
Comment (AnthonyLaw) [reply __tests__/infrastructure/RestrictionService.spec.js:88]: yes, I just want to add all MosaicRestrictionType into the test.

**Comment by AnthonyLaw** [comment]
> @AnthonyLaw I got a question regarding global restriction. You added (on beginning) mosaicId to the "Restrictions" column:
> 
> image
> 
> it's possible that we will have another mosaicId that we currently looking at the page?

yes, based on the API, the `restrictions` returns in Array.  

image

**Review by Jaguar0625 — APPROVED** [review]

File: __tests__/infrastructure/RestrictionService.spec.js (hunk 2 above)
Comment (Jaguar0625) [reply __tests__/infrastructure/RestrictionService.spec.js:88]: i mean if you wanted to make the names more distinct, as "greater|less than or equal" would be more proper.

**Pull request merged**